analyzers/src/SonarAnalyzer.Common/Rules/Utilities/SymbolReferenceAnalyzerBase.cs

```diff
@@ -33,40 +34,25 @@ public abstract class SymbolReferenceAnalyzerBase<TSyntaxKind> : UtilityAnalyzer
         private const string Title = "Symbol reference calculator";
         private const int TokenCountThreshold = 40_000;
 
-        private readonly ISet<SymbolKind> declarationKinds = new HashSet<SymbolKind>
-        {
-            SymbolKind.Event,
-            SymbolKind.Field,
-            SymbolKind.Local,
-            SymbolKind.Method,
-            SymbolKind.NamedType,
-            SymbolKind.Parameter,
-            SymbolKind.Property,
-            SymbolKind.TypeParameter
-        };
+        protected sealed override string FileName => "symrefs.pb";
 
         protected abstract SyntaxNode GetBindableParent(SyntaxToken token);
 
-        protected sealed override string FileName => "symrefs.pb";
+        protected abstract IEnumerable<ReferenceInfo> CreateDeclarationReferenceInfo(SyntaxNode node, SemanticModel model);
+
+        protected abstract IEnumerable<SyntaxNode> GetDeclarations(SyntaxNode node);
+
+        protected abstract StringComparer IdentifierComparer { get; }
```

Comment:
And I suggest to add a UT for visual basic that makes sure we use the correct comparison.
The same identifier but with different casing UT.

Reply:
Maybe I'm missing something but I cannot use `Language.NameComparision` since it's type is `StringComparison` and the `HashSet` requires an `IEqualityComparer<T>`.

Reply:
Actually I will extend `ILanguageFacade` to have also a `StringComparer`

Reply:
Ah, those two comparer/comparison pairs. 
Please add `StringComparer NameComparer { get; }` to `ILanguageFacade` the same way as `NameComparison` is there. It will surely be helpful on other places.

Reply:
Yep :D